src/app/utilities/AnnotateUtilities.ts

```diff
@@ -2,17 +2,70 @@
 import jp from 'jsonpath';
 
 /* Import Utilities */
-import { ExtractFromSchema, ExtractClassesAndTermsFromSchema } from 'app/utilities/SchemaUtilities';
+import { ExtractFromSchema, ExtractClassesAndTermsFromSchema, MakeJsonPathReadableString } from 'app/utilities/SchemaUtilities';
 
 /* Import Types */
 import { DigitalSpecimen } from "app/types/DigitalSpecimen";
 import { DigitalMedia } from "app/types/DigitalMedia";
-import { AnnotationFormProperty, Dict } from "app/Types";
+import { AnnotationFormProperty, AnnotationTarget, ParentClass, Dict } from "app/Types";
 
 
 /* Utilities associated with annotating */
 
 
+/**
+ * Function to extract parent classes from a given super class and provide all their associated information
+ * @param params An object containing: the annotationTarget, superClass and parentClasses properties
+ * @returns The extracted parent classes
+ */
+const ExtractParentClasses = (params: {
+    annotationTarget: AnnotationTarget,
+    superClass: DigitalSpecimen | DigitalMedia | Dict
+}) => {
+    const { annotationTarget, superClass } = params;
+
+    /* Base variables */
+    const pathArray: string[] = ['$'];
+    const parents = jp.parse(annotationTarget.jsonPath).slice(1, -1);
+    const parentClasses: ParentClass[] = [];
+
+    /* For each parant, find and extract its class information */
+    parents.forEach((parent, index) => {
+        /* Filter path from schemas */
+        let parentName = parent.expression.value.replace('ods:', '').replace('dwc:', '').replace('dcterms:', '');
+
+        /* Add to pathArray */
+        pathArray.push(parent.expression.value);
+
+        if (parentName.includes('has') || parentName[0] === parentName[0].toUpperCase()) {
+            /* If a parent class is empty, it needs to be targeted, if it is not completely empty, check for indexes */
+            const parentNodes = jp.nodes(superClass, jp.stringify(pathArray).replaceAll('][', ']..[').replaceAll('"', "'"));
```

Comment:
`][', ']..[` is that correct? does `..` mean something special in javascript?

Reply:
That is something from the jsonPath library. When you provide two dots it will try to search and target any matching nodes within an object that align with the provided JSON path

Reply:
:+1: nice